### PR TITLE
Add RBAC for the flowcontrol.apiserver.k8s.io

### LINKED ISSUE #1430: Fail to list APF resources when the Visibility API is turned on
**What happened**:

The following messages are repeatedly emitted in the operator logs: 

```
W1208 15:30:36.007289       1 reflector.go:535] pkg/mod/k8s.io/client-go@v0.28.4/tools/cache/reflector.go:229: failed to list *v1beta3.PriorityLevelConfiguration: prioritylevelconfigurations.flowcontrol.apiserver.k8s.io is forbidden: User "system:serviceaccount:kueue-system:kueue-controller-manager" cannot list resource "prioritylevelconfigurations" in API group "flowcontrol.apiserver.k8s.io" at the cluster scope
E1208 15:30:36.007354       1 reflector.go:147] pkg/mod/k8s.io/client-go@v0.28.4/tools/cache/reflector.go:229: Failed to watch *v1beta3.PriorityLevelConfiguration: failed to list *v1beta3.PriorityLevelConfiguration: prioritylevelconfigurations.flowcontrol.apiserver.k8s.io is forbidden: User "system:serviceaccount:kueue-system:kueue-controller-manager" cannot list resource "prioritylevelconfigurations" in API group "flowcontrol.apiserver.k8s.io" at the cluster scope

W1208 15:30:40.237300       1 reflector.go:535] pkg/mod/k8s.io/client-go@v0.28.4/tools/cache/reflector.go:229: failed to list *v1beta3.FlowSchema: flowschemas.flowcontrol.apiserver.k8s.io is forbidden: User "system:serviceaccount:kueue-system:kueue-controller-manager" cannot list resource "flowschemas" in API group "flowcontrol.apiserver.k8s.io" at the cluster scope
E1208 15:30:40.237370       1 reflector.go:147] pkg/mod/k8s.io/client-go@v0.28.4/tools/cache/reflector.go:229: Failed to watch *v1beta3.FlowSchema: failed to list *v1beta3.FlowSchema: flowschemas.flowcontrol.apiserver.k8s.io is forbidden: User "system:serviceaccount:kueue-system:kueue-controller-manager" cannot list resource "flowschemas" in API group "flowcontrol.apiserver.k8s.io" at the cluster scope
```

**How to reproduce it (as minimally and precisely as possible)**:

Install the Visibility API service and start the operator with `--feature-gates=VisibilityOnDemand=true`.

**Anything else we need to know?**:

APF is enabled by default when creating a generic API server. It could either be disabled, or the required RBAC be added.

**Environment**:
- Kueue version (use `git describe --tags --dirty --always`): `v0.5.0-devel-396-g6d428f3-dirty`

### PR DESCRIPTION
**Pull request by tenzen-y**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
Since we use the API Aggregation layer on the VisibilityOnDemand, the kueue-manager needs to have permission to list flowschemas and prioritylevelconfigurations.

#### Which issue(s) this PR fixes:

Fixes #1430

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -111,7 +111,7 @@ manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and Cust
 	$(CONTROLLER_GEN) \
 		rbac:roleName=manager-role output:rbac:artifacts:config=config/components/rbac\
 		webhook output:webhook:artifacts:config=config/components/webhook\
-		paths="./pkg/controller/...;./pkg/webhooks/...;./pkg/util/cert/..."
+		paths="./pkg/controller/...;./pkg/webhooks/...;./pkg/util/cert/...;./pkg/visibility/..."
```
Comment (mimowo) [inline Makefile:114]: This seems to be breaking the build. I remember there was an issue with including the `/pkg/visibility` from scanning. Maybe we could move the markers to the `/apis/visibility`, or scan just a sub-package of `pkg/visibility` which contains the markers.

Comment (tenzen-y) [reply Makefile:114]: I'll put these markers on `/pkg/visibility/api`. That should work fine.

Comment (mimowo) [reply Makefile:114]: nice, thanks for investigating

**Pull request merged**